### ISSUE TITLE
Node.js 14 - sporadic npm failures on x86

### ISSUE THREAD

**Issue by kewde** [appveyor/ci#3407]
@FeodorFitsner 

I've been having a consistent failure *specifically* with Node `v14` on `x86`. (note x64 works fine!)
https://github.com/mapbox/node-sqlite3/issues/1318

`npm` is acting up a lot, the failure is consistent but the locus of the failure is very inconsistent.
Sometimes even calling `npm -v` will return error `-1073741819`, next run of the CI is returns fine but fails a bit later at another call of `npm`..

It's as if the setting of the permissions on `npm` happens async and decides to lock out it out at some point.

**Comment by kewde** [comment]
I found the culprit I think.
```
powershell Set-ExecutionPolicy Unrestricted -Scope CurrentUser -Force
IF %ERRORLEVEL% NEQ 0 GOTO ERROR
```

**Comment by kewde** [comment]
After further inspection, it does remain unstable.
One time the CI worked fine: https://ci.appveyor.com/project/Mapbox/node-sqlite3/builds/32511420

One commit later, which re-enabled some of the informational npm requests seems to have undone the change. Perhaps `npm root -g` sporadically triggers some access control mechanism due to the fact that it's global? It wouldn't explain the failures on `npm -v` though..

One theory is that commands like `npm root`, `npm bin` etc  take up time, removing them from the execution perhaps delayed the occurrence of the error. 

It's a very strange bug.

**Comment by FeodorFitsner** [comment]
So, you mean calling `npm -v` in a loop would be a repro for this issue?

**Comment by kewde** [comment]
@FeodorFitsner 
I've made a repro for you.

**os**: Visual Studio 2015
The AppVeyor config
https://github.com/kewde/node-sqlite3/blob/kewde/repro-appveyor-ci/appveyor.yml
The script it launches with the simple loop (10k iterations of `npm -v`).
https://github.com/kewde/node-sqlite3/blob/kewde/repro-appveyor-ci/scripts/build-appveyor.bat
The results of the CI
https://ci.appveyor.com/project/Mapbox/node-sqlite3/builds/32524770

**Comment by kewde** [comment]
It's not a bug in `npm` because I don't experience these failures using the same version (6.14.4) on Node v13.

**Comment by kewde** [comment]
It's definitely a bug that's caused by the combined use of Node v14 & npm on x86.
I've attempted to replicate the permission issue using just a loop of `node -v` which didn't seem to cause any trouble.

**Comment by FeodorFitsner** [comment]
Maybe it's better to submit an issue upstream (Node/NPM)?

**Comment by kewde** [comment]
@FeodorFitsner I have come to the same conclusion, I was hoping perhaps that this might just have been a permission issue on AppVeyors end.

**Comment by FeodorFitsner** [comment]
Funny enough the problem doesn't exist anymore in Node 14.1.0: https://ci.appveyor.com/project/FeodorFitsner/node-14-x86-issue/build/job/5dvx1arrlya8eh27

**Comment by kewde** [comment]
I'm glad to hear that, that should resolve my issue as the ABI remains the same over minor versions.

**Comment by vweevers** [comment]
FYI, this also happens on 14.3.0, but not 14.2.0.

Appears to be a bug in node, an access violation in V8. If I'm able to confirm (AppVeyor's RDP helped but I'm hoping to replicate locally) then I'll open an issue there.

**Comment by FeodorFitsner** [comment]
@vweevers, @kewde have you ever submitted a bug to Node regarding that (or would you able to help us to file/describe the issue enough for Node team to consider it)? Looks like it's back again in 14.4.0 - we've got some new reports. Thank you!

**Comment by vweevers** [comment]
@FeodorFitsner I was unable to reproduce locally. Here's how far I got on an appveyor machine within the RDP session's time limit, with node 14.3.0 x86. If I remember correctly I ran (the equivalent of) `npm -v`.

![appveyor access violation 2](https://user-images.githubusercontent.com/3055345/86400744-72d0c180-bca9-11ea-893b-195e6060212d.png)

**Comment by FeodorFitsner** [comment]
We can enable a longer timeout for you if you are willing to help further and possibly file an issue?

**Comment by erossignon** [comment]
This happened in https://ci.appveyor.com/project/erossignon/node-opcua-crypto  and https://ci.appveyor.com/project/erossignon/node-opcua-pki too node-js with nodejs 14.2.0 x86.

**Comment by FeodorFitsner** [comment]
Use x64 version of Node which doesn't have the issue.

**Comment by sidneys** [comment]
Encountered this issue on v14.8.0 as well.

**Comment by erossignon** [comment]
> Use x64 version of Node which doesn't have the issue.

Yes I am using x64 , but I also need x32. The purpose of using AppVeyor in my case it to test that the software is working on a wide variety of architecture, and nodejs version ...

**Comment by FeodorFitsner** [comment]
Could you guys do a simplest repro for this issue please?

**Comment by FeodorFitsner** [comment]
It looks like node/npm bug. We've been able to reproduce the issue on various platforms and bare-metal PC with Windows 10: https://github.com/nodejs/node/issues/34927

**Comment by bitjson** [comment]
If it's helpful, this issue is reliably affecting [`typescript-starter`](https://github.com/bitjson/typescript-starter) AppVeyor builds, e.g. 
[build 710](https://ci.appveyor.com/project/bitjson/typescript-starter/build/job/e05www6ioa0shhob).

**Comment by FeodorFitsner** [comment]
Use x64 version of Node.

**Comment by bitjson** [comment]
> Use x64 version of Node.

Ah, I hadn't looked at the AppVeyor configuration in a while, [thanks](https://github.com/bitjson/typescript-starter/commit/5a84c78f3141947a17769efee87addcc2057d6da)!
